src/plugins/discord/fetcher.test.js

```diff
@@ -0,0 +1,299 @@
+  const defaultOptions = () => ({
+    membersLimit: 100,
+    messagesLimit: 100,
+    reactionsLimit: 100,
+  });
+
+  describe("fetch guild", () => {
+    it("passes correct endpoint", async () => {
+      const fetch = jest.fn(() => Promise.resolve([]));
+      const fetcher = new Fetcher(fetch, defaultOptions());
+      await fetcher.guild("1");
+      expect(fetch.mock.calls[0]).toEqual(["guilds/1"]);
+    });
+
+    it("handles response", async () => {
+      const expected: Guild = {id: "1", name: "guildname", permissions: 0};
+      const fetch = jest.fn(() => Promise.resolve(expected));
+      const fetcher = new Fetcher(fetch, defaultOptions());
+      const guild = await fetcher.guild("1");
+      expect(guild).toEqual(expected);
+    });
+  });
+
+  describe("fetch channels", () => {
+    it("passes correct endpoint", async () => {
+      const fetch = jest.fn(() => Promise.resolve([]));
+      const fetcher = new Fetcher(fetch, defaultOptions());
+      await fetcher.channels("1");
+      expect(fetch.mock.calls[0]).toEqual(["/guilds/1/channels"]);
+    });
+
+    it("handles response", async () => {
+      // response object from server, prior to type transformation
+      const testChannel: any = (id: string) => ({
+        id: id,
+        name: "testChannelName",
+        type: 0,
+      });
+
+      const testChannelObj = (id: Snowflake): Channel => ({
+        id: id,
+        name: "testChannelName",
+        type: "GUILD_TEXT",
+      });
+
+      const response: any = [testChannel("1")];
+      const expected: Channel[] = [testChannelObj("1")];
+      const fetch = jest.fn(() => Promise.resolve(response));
+      const fetcher = new Fetcher(fetch, defaultOptions());
+      const data = await fetcher.channels("0");
+      expect(data).toEqual(expected);
+    });
+  });
+
+  describe("fetch members", () => {
+    const testMember: any = (userId: string) => ({
+      user: {
+        id: userId,
+        username: "username",
+        discriminator: "disc",
+        bot: true,
+      },
+      nick: "nickname",
+      roles: ["test role"],
+    });
+
+    const options = () => ({
+      membersLimit: 2,
+      messagesLimit: 100,
+      reactionsLimit: 100,
+    });
+
+    it("passes correct endpoint", async () => {
+      const fetch = jest.fn(() => Promise.resolve([]));
+      const fetcher = new Fetcher(fetch, options());
+      await fetcher.members("1", "0");
+      expect(fetch.mock.calls[0]).toEqual([
+        "/guilds/1/members?after=0&limit=2",
+      ]);
+    });
+
+    it("handles response", async () => {
+      const response: GuildMember[] = [testMember("1")];
+      const fetch = jest.fn(() => Promise.resolve(response));
+      const fetcher = new Fetcher(fetch, options());
+      const {results} = await fetcher.members("1", "0");
+      expect(results).toEqual(response);
+    });
+
+    it("returns correct endCursor", async () => {
+      const response: any = [testMember("1"), testMember("2")];
```

Comment:
We are losing type information here as well.
"Downgrading" from `any[]` to `any`.

For consistency, let's use `any[]`.

Reply:
updated